src/transformers/pipelines.py

```diff
@@ -396,9 +403,12 @@ def __init__(
             self.model = self.model.to(self.device)
 
         # Update config with task specific parameters
+        if self.task not in SUPPORTED_TASKS and self.abstract_task not in SUPPORTED_TASKS:
+            raise KeyError("Unknown task {}, available tasks are {}".format(self.task, list(SUPPORTED_TASKS.keys())))
```

Comment:
I guess I don't see the relation between `SUPPORTED_TASKS` (which is a list of shortcuts to a specific subset of demo pipelines) and the implementation of the generic `Pipeline`. At a minimum I would remove those two lines.

Reply:
I don't like the fact that there's a bidirectional relationship between the list of shortcuts (which I don't use, because I know a model I want to use is e.g. a text-classification model, etc.) and the actual implementations

Reply:
Agree! 

I think it's a good idea to separate these two names so that we have:

1) A generic task name for each Pipeline, *e.g.*: "feature-extraction", "translation", "text-classification" (not sentiment-analysis). Those names will be used to look up the correct configuration of the model. They can be thought of as something similar to the `base_model_prefix` for the models

2) A more specific task  name Mapping which is used to have a good default value when loading a specific pipeline. Here we will use "ner" and "sentiment-analysis".

Does that sound better? @julien-c

Reply:
As discussed offline, let's wait a little before merging this, to see how the dust settles:)